### PR TITLE
Doxygen docs for mbed-hal - round 1

### PR DESCRIPTION
**Pull request by 0xc0170**
Round 1 , captures the basic info

@bogdanm @niklas-arm @bremoran

### CODE REVIEW

File: mbed-hal/gpio_irq_api.h [hunk 1]
```diff
@@ -24,22 +24,66 @@
...
+ * This is target dependent, as it might enable entire port or just a pin
+ * @param obj The gpio object
+ */
 void gpio_irq_enable(gpio_irq_t *obj);
+
+/** Disable IRQ
+ *
+ * This is target dependent, as it might disable entire port or just a pin
+ * @param obj The gpio object
+ * @param pin The gpio pin
```
Comment (niklas-arm) [inline mbed-hal/gpio_irq_api.h]: were is that pin? ;-)

**Comment by niklas-arm** [comment]
Do we need to document the `obj` argument every single time?
It's a coding convention to do OOP in C, should only be documented at the high level.

**Comment by niklas-arm** [comment]
There are also other redundant descriptions like `@param pin The gpio pin`.

**Comment by 0xc0170** [comment]
> Do we need to document the obj argument every single time?

I would vote for function parameters - all or none. Others?

**Comment by 0xc0170** [comment]
cc @iriark01 - proof read please

**Comment by 0xc0170** [comment]
I did more corrections. Irit will send corrections via separate PR.

Please review, I would like to merge this this week

**Pull request merged**